### ISSUE TITLE
value gives [object Object] if inputComponnent={Input} and this Input is imported from "@material-ui/core"

### ISSUE THREAD

**Issue by mghazanfar** [catamphetamine/react-phone-number-input#214]
```
<PhoneInput
              inputComponent={TextField}
              limitMaxLength
              value={phone}
              onChange={value => this.handlePhone(value)}
              error={phone ? (isValidPhoneNumber(phone) ? undefined : 'Invalid phone number') : 
'Phone number required'}
 />
```
Neither TextField not Input from material-ui/core works fine. Your component works perfectly fine but when we use Input from material-ui, it starts giving [object Object] in value where expected value is the entered phone number. Please help!

**Comment by mghazanfar** [comment]
And also, using /react-responsive-ui for select to look appropriate for my app, I'm unable to type and search my desired country.

**Comment by mghazanfar** [comment]
@catamphetamine I inspected this. This is the value coming fromm onChange function and this is what comes in value and state is set.

**Comment by mghazanfar** [comment]
Any workaround till now @catamphetamine

**Comment by catamphetamine** [comment]
> And also, using /react-responsive-ui for select to look appropriate for my app, I'm unable to type and search my desired country.

This feature is not supported in `react-responsive-ui`.
You can write adapter for a custom Select using `react-responsive-ui` example.

> I inspected this. This is the value coming from onChange function and this is what comes in value and state is set.

Post what exactly is the value, which properties does it have, etc.

**Comment by TylerRick** [comment]
I ran into something similar when I tried to integrate this with `TextField` from `@material-ui/core`.

I think the problem was the that `PhoneInput`'s onChange handler is expected to return a string value (phone) but `MuiTextField` onChange returns an event:

`onChange={(event) => {return event.target.value}}`

So as soon as you type, it caused an error.

There's probably a way to write an adapter but I gave up and just used the default input component for now...

**Comment by catamphetamine** [comment]
@TylerRick 
> There's probably a way to write an adapter but I gave up and just used the default input component for now...

If the input field expects an `event` then one could simulate it with something like:

```js
<CustomInput
  onChange={value => onChange({
    target: {
      value
    }
  )} />
```

Or something like that (maybe with dummy functions like `preventDefault`).
(Google something like "simulate input event from value")

**Comment by TylerRick** [comment]
Thank you so much for your reply and suggested workaround. I don't think we even need to simulate an event though.

I finally found the solution to this problem (getting `[object Object]` as the value) that @mghazanfar 
 and I were having, in the comments on #226: @NevenLiang had a good [example](https://github.com/catamphetamine/react-phone-number-input/issues/226#issuecomment-511300590) of an adapter component that adapts between the API of TextField and the API required by PhoneInput's `inputComponent`, which solves this and a few other issues (like the need to implement `input.focus`)

I've created a sandbox demo here: https://codesandbox.io/s/integrate-react-phone-number-input-with-material-uicore-ohfwp

The next major pain point with using material-ui is that there doesn't appear to be a way to pass props down to the input component. I'm suggesting we add one in issue #280.

-----

> Inspect the [object Object] and see what it is.

It's a string. Something is coercing the Event object into the string, `"[object Object]"`.

**Comment by catamphetamine** [comment]
@TylerRick 

> I finally found the solution to this problem (getting [object Object] as the value) that @mghazanfar
and I were having, in the comments on #226

And I finally figured out that indeed `inputComponent` isn't what I initially thought it was when replying in this issue.
Lol.
Indeed, just swapping `inputComponent` isn't enough.
I've added a new `numberInputComponent` property — this one is swappable and could be set to a Material UI input or whatever else.

FYI: Released `react-phone-number-input@2.3.25` with the `numberInputComponent` property.
